src/MCPClient/lib/clientScripts/extract_contents.py

```diff
@@ -145,39 +191,52 @@ def main(job, transfer_uuid, sip_directory, date, task_uuid, delete=False):
 
         if not exitstatus == 0:
             # Dang, looks like the extraction failed
-            job.pyprint('Command', command.description, 'failed!', file=sys.stderr)
+            job.pyprint('Command',
+                        command.description,
+                        'failed!',
+                        file=sys.stderr)
         else:
             extracted = True
-            job.pyprint('Extracted contents from', os.path.basename(file_path))
+            job.pyprint('Extracted contents from',
+                        os.path.basename(output_file_path))
 
             # Assign UUIDs and insert them into the database, so the newly-
             # extracted files are properly tracked by Archivematica
-            extracted_path = output_directory(file_path, date)
-            for extracted_file in tree(extracted_path):
-                assign_uuid(job,
-                            extracted_file, file_.uuid, transfer_uuid, date,
-                            task_uuid, sip_directory, file_.currentlocation)
-
-            # Assign UUIDs to directories via ``Directory`` objects in the
-            # database.
+            for extracted_file in tree(output_file_path):
+                assign_uuid(
+                    job, extracted_file, file_.uuid, transfer_uuid, date,
+                    task_uuid, sip_directory, file_.currentlocation,
+                    file_.originallocation, output_file_path)
+
             if transfer_mdl.diruuids:
-                Directory.create_many(
-                    get_dir_uuids(
-                        _get_subdir_paths(extracted_path, sip_directory),
-                        logger,
-                        printfn=job.pyprint),
-                    transfer_mdl)
-
-            # We may want to remove the original package file after extracting its contents
+                create_extracted_dir_uuids(
+                    job, transfer_mdl, output_file_path, sip_directory, file_)
+
+            # We may want to remove the original package file after extracting
+            # its contents
             if delete:
-                delete_and_record_package_file(job, file_path, file_.uuid, file_.currentlocation)
+                delete_and_record_package_file(
+                    job, temp_dir, file_.uuid, file_.currentlocation)
 
     if extracted:
         return 0
     else:
         return 255
 
 
+def create_extracted_dir_uuids(
+        job, transfer_mdl, output_file_path, sip_directory, file_):
+    # Assign UUIDs to directories via ``Directory`` objects in the
+    # database.
```

Comment:
Because this is now a function, the comment should be a proper docstring.

Reply:
:+1: